apis/outlook/calendar/code/main.go

```diff
@@ -0,0 +1,146 @@
+			os.Exit(1)
+		}
+	case "listEventsToday":
+		now := time.Now()
+		start := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, now.Location())
+		end := time.Date(now.Year(), now.Month(), now.Day(), 23, 59, 59, 0, now.Location())
+		if err := commands.ListEvents(context.Background(), start, end); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "listEvents":
+		start, end, err := parseStartEnd(os.Getenv("START"), os.Getenv("END"), false)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		if err := commands.ListEvents(context.Background(), *start, *end); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "getEventDetails":
+		if err := commands.GetEventDetails(context.Background(), os.Getenv("EVENT_ID"), os.Getenv("CALENDAR_ID"), graph.OwnerType(os.Getenv("OWNER_TYPE"))); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "createEvent":
+		info := graph.CreateEventInfo{
+			Attendees: strings.Split(os.Getenv("ATTENDEES"), ","),
+			Subject:   os.Getenv("SUBJECT"),
+			Location:  os.Getenv("LOCATION"),
+			Body:      os.Getenv("BODY"),
+		}
+
+		isOnline, err := strconv.ParseBool(os.Getenv("IS_ONLINE"))
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		info.IsOnline = isOnline
+
+		if id := os.Getenv("CALENDAR_ID"); id != "" {
+			info.ID = id
+			info.Owner = graph.OwnerType(os.Getenv("OWNER_TYPE"))
+		}
+
+		start, end, err := parseStartEnd(os.Getenv("START"), os.Getenv("END"), false)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		info.Start = *start
+		info.End = *end
+
+		if err := commands.CreateEvent(context.Background(), info); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "inviteUserToEvent":
+		if err := commands.InviteUserToEvent(context.Background(), os.Getenv("EVENT_ID"), os.Getenv("CALENDAR_ID"), graph.OwnerType(os.Getenv("OWNER_TYPE")), os.Getenv("USER_EMAIL"), os.Getenv("MESSAGE")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "deleteEvent":
+		if err := commands.DeleteEvent(context.Background(), os.Getenv("EVENT_ID"), os.Getenv("CALENDAR_ID"), graph.OwnerType(os.Getenv("OWNER_TYPE"))); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "searchEvents":
+		start, end, err := parseStartEnd(os.Getenv("START"), os.Getenv("END"), false)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		if err := commands.SearchEvents(context.Background(), os.Getenv("QUERY"), *start, *end); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "respondToEvent":
+		if err := commands.RespondToEvent(context.Background(), os.Getenv("EVENT_ID"), os.Getenv("CALENDAR_ID"), graph.OwnerType(os.Getenv("OWNER_TYPE")), os.Getenv("RESPONSE")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	default:
+		fmt.Printf("Unknown command: %q\n", command)
+		os.Exit(1)
+	}
+}
+
+func parseStartEnd(start, end string, optional bool) (*time.Time, *time.Time, error) {
+	var (
+		startPtr *time.Time
+		endPtr   *time.Time
```

Comment:
Typically, pointers aren't used for `time.Time`. Sort of why there is a `time.Time.IsZero()` method.

Reply:
Fixed